nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureBlobProcessor_v12.java

```diff
@@ -113,6 +131,59 @@ protected BlobServiceClient getStorageClient() {
         return storageClient;
     }
 
+    protected BlobClient getBlobClient(PropertyContext context, BlobContainerClient containerClient, String blobName) throws DecoderException {
+        final String cseKeyTypeValue = context.getProperty(AzureBlobClientSideEncryptionUtils_v12.CSE_KEY_TYPE).getValue();
+        final AzureBlobClientSideEncryptionMethod_v12 cseKeyType = AzureBlobClientSideEncryptionMethod_v12.valueOf(cseKeyTypeValue);
+
+        final String cseKeyId = context.getProperty(AzureBlobClientSideEncryptionUtils_v12.CSE_KEY_ID).getValue();
+
+        final String cseLocalKeyHex = context.getProperty(AzureBlobClientSideEncryptionUtils_v12.CSE_LOCAL_KEY_HEX).getValue();
+
+        BlobClient blobClient = containerClient.getBlobClient(blobName);
+
+        if (cseKeyType == AzureBlobClientSideEncryptionMethod_v12.LOCAL) {
+            byte[] keyBytes = Hex.decodeHex(cseLocalKeyHex.toCharArray());
```

Comment:
The `toCharArray()` is unnecessary.
```suggestion
            byte[] keyBytes = Hex.decodeHex(cseLocalKeyHex);
```